src/authz/role.guard.ts

```diff
@@ -19,10 +20,17 @@ export class RolesGuard implements CanActivate {
   async canActivate(context: ExecutionContext): Promise<boolean> {
     const req = context.switchToHttp().getRequest();
     const roles = this.reflector.get<string[]>(ROLES_METADATA_KEY, context.getHandler());
+    let userRole: string;
+
     if (!roles) {
       return true;
     }
 
+    if (req.headers.authorization === process.env.INTERNAL_SECRET) {
+      userRole = Role.Internal;
+      return this.matchRoles(roles, userRole)
```

Comment:
אם עושים מיד return, אין טעם לאתחל את המשתנה userRole

Reply:
האם זה לא קריא יותר ככה?, מאשר:
    if (req.headers.authorization === process.env.INTERNAL_SECRET) {
      return this.matchRoles(roles, Role.Internal)
    }
אם כדאי אני אעלה את השינוי

Reply:
אני חושבת שאין טעם במשתנה הזה

Reply:
יש בו שימוש, בהמשך.
השאלה אם פה ההשמה מיותרת.